### PR TITLE
(FACT-911, FACT-915) Fix dmi facts on sparc

### PR DESCRIPTION
**Pull request by MikaelSmith**
Fixes manufacturer and productname facts, and implements the serialnumber fact a la 2.x.

### CODE REVIEW

File: lib/src/facts/solaris/dmi_resolver.cc [hunk 1]
```diff
@@ -65,20 +66,27 @@ namespace facter { namespace facts { namespace solaris {
                 }
                 return result.chassis_type.empty() || result.chassis_asset_tag.empty();
             });
-        } else if (arch && arch->value() == "sparc") {
-            static boost::regex line_re("System Configuration: (.+) sun\\d.");
+        } else if (isa && isa->value() == "sparc") {
+            static boost::regex manufacturer_rx("^System Configuration: (.+) sun\\d.$");
+            static boost::regex productname_rx("^SUNW,(.*)$");
             // prtdiag is not implemented in all sparc machines, so we cant get product name this way.
+            int found = 0;
```
Comment (MikaelSmith) [inline lib/src/facts/solaris/dmi_resolver.cc]: Going to rework this to be a little more clear.

Comment (MikaelSmith) [inline lib/src/facts/solaris/dmi_resolver.cc]: Updated it to use the pattern established in other loops in the file.

**Comment by MikaelSmith** [comment]
```
$ /opt/puppetlabs/bin/facter dmi os
dmi => {
  manufacturer => "Oracle Corporation",
  product => {
    name => "SPARC-Enterprise-T5120"
  }
}
os => {
  architecture => "sun4v",
  family => "Solaris",
  hardware => "sun4v",
  name => "Solaris",
  release => {
    full => "10.0",
    major => "10",
    minor => "0"
  }
}
```

**Comment by whopper** [comment]
:+1: also pre-FR'd this on Solaris 10 SPARC. Looks good.

**Pull request merged**